src/analytics/formResponseTracker/formResponseTracker.test.ts

```diff
@@ -0,0 +1,204 @@
+import { describe, expect, jest, test } from "@jest/globals";
+import { FormResponseTracker } from "./formResponseTracker";
+import { FormEventInterface } from "../formTracker/formTracker.interface";
+window.DI = { analyticsGa4: { cookie: { consent: true } } };
+
+describe("FormResponseTracker", () => {
+  const action = new Event("submit", {
+    bubbles: true,
+    cancelable: true,
+  });
+
+  const spy = jest.spyOn(FormResponseTracker.prototype, "pushToDataLayer");
+  const constructorSpy = jest.spyOn(
+    FormResponseTracker.prototype,
+    "initialiseEventListener",
+  );
+
+  test("new instance should call initialiseEventListener", () => {
+    const instance = new FormResponseTracker();
+    expect(instance.initialiseEventListener).toBeCalled();
+  });
+});
+
+describe("form with input checkbox", () => {
+  const action = new Event("submit", {
+    bubbles: true,
+    cancelable: true,
+  });
+
+  const spy = jest.spyOn(FormResponseTracker.prototype, "pushToDataLayer");
+
+  test("datalayer event should be defined", () => {
+    const instance = new FormResponseTracker();
+    document.body.innerHTML = "";
```

Comment:
Do we need the `document.body.innerHTML = "";` line aswell as the below definition?

Reply:
removed